pymodbus/pdu/pdu.py

```diff
@@ -128,7 +142,8 @@ def __init__(
         super().__init__(transaction_id=transaction, dev_id=device_id)
         self.function_code = function_code | 0x80
         self.exception_code = exception_code
-        Log.error(f"Exception response {self.function_code} / {self.exception_code}")
+        self.exception_name = self.exception_code_dict.get(self.exception_code, f"Unknown Exception value: {self.exception_code}")
+        Log.error(f"Exception response FC{self.function_code & 0x7F} / {self.exception_name}")
```

Comment:
no need to add 0x7f just use the function_code parameter.

Just showing the name, makes it more difficult for log parsers so the code should still be presented.

Showing the exception name makes sense for people who do not know the modbus standard, but the same people to not know the function codes either....so it would be better to either use names for both or not at all.

Reply:
Ok, so which approach here will be the best?

Reply:
make \_\_str\_\_

Reply:
> in general instead of logging, it would be better as \_\_str\_\_ allowing the app to do what it want.

For now __str__ only shows base pdu variables. Which can be confusing if app will use __str__ on exception.

Reply:
And with the log?

Reply:
no